File: src/main/java/com/github/svegon/capi/mixininterface/IAbstractSignEditScreen.java
package com.github.svegon.capi.mixininterface;

import net.minecraft.block.WoodType;
import net.minecraft.block.entity.SignBlockEntity;
import net.minecraft.block.entity.SignText;
import net.minecraft.client.util.SelectionManager;
import org.jetbrains.annotations.Nullable;

public interface IAbstractSignEditScreen {
    SignBlockEntity getBlockEntity();

    void setBlockEntity(SignBlockEntity blockEntity);

    SignText getSignText();

    void setSignText(SignText text);

    String[] getText();

    void setText(String... text);

    boolean front();

    void front(boolean front);

    WoodType getSignType();

    void setSignType(WoodType signType);

    int currentRow();

    void currentRow(int row);

    @Nullable
    SelectionManager getSelectionManager();

    void setSelectionManager(@Nullable SelectionManager selectionManager);
}

File: src/main/kotlin/io/github/svegon/mclientapi/mixininterface/IAbstractSignEditScreen.kt
package io.github.svegon.capi.mixininterface

import net.minecraft.client.util.SelectionManager

interface IAbstractSignEditScreen {
    fun getBlockEntity(): SignBlockEntity?

    fun setBlockEntity(blockEntity: SignBlockEntity?)

    fun getSignText(): SignText?

    fun setSignText(text: SignText?)

    var text: Array<String?>?

    fun front(): Boolean

    fun front(front: Boolean)

    fun getSignType(): WoodType?

    fun setSignType(signType: WoodType?)

    fun currentRow(): Int

    fun currentRow(row: Int)

    fun getSelectionManager(): SelectionManager?

    fun setSelectionManager(selectionManager: SelectionManager?)
}
